### PR TITLE
Merge pull request #9 from kouch-chivorn/create-categories

### PR DESCRIPTION
**Pull request by kouch-chivorn**
create categories for admin

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
+    field = f.hidden_field(:_destroy) 
+    link = link_to label,"#", onclick: "remove_fields(this)", remote: true
+    field + link
+  end
+  
+  def link_to_add_fields(name, f, association)
+    new_object = f.object.class.reflect_on_association(association).klass.new
+    fields = f.fields_for(association, new_object, 
+      child_index: "new_#{association}") do |builder|
+      render(association.to_s.singularize + "_fields", :f => builder)
```
Comment (dieunb) [inline app/helpers/application_helper.rb]:
```
render(association.to_s.singularize + "_fields", f: builder)
```

File: app/controllers/admins/questions_controller.rb [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    flash[:notice] = t "question.message.delete", question: @question.name 
+    redirect_to [:admins, @category, :questions]
+  end
+
+  private
+  def set_question
+    @question = Question.find params[:id]
+  end
+
+  def set_category
```
Comment (dieunb) [inline app/controllers/admins/questions_controller.rb]: I don't see you use it

File: app/controllers/admins/questions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    else
+      render :edit
+    end
+  end
+
+  def show
+  end
+
+  def edit
+    @word.category = @category
```
Comment (dieunb) [inline app/controllers/admins/questions_controller.rb]: `@word`???

File: app/helpers/application_helper.rb [hunk 4]
```diff
@@ -1,18 +1,33 @@
...
   end
 
   def devise_mapping
-    @devise_mapping ||= Devise.mappings[:user] || Devise.mappings[:admin]
+    @devise_mapping ||= Devise.mappings[:user]
   end
 
-  def shorten str
-    str
+  def shorten(description)
```
Comment (dieunb) [inline app/helpers/application_helper.rb]: You can read [this ducument](http://apidock.com/rails/ActionView/Helpers/TextHelper/truncate) to use

File: app/models/question.rb [hunk 5]
```diff
@@ -2,10 +2,18 @@ class Question < ActiveRecord::Base
...
   accepts_nested_attributes_for :answers, allow_destroy: true, 
                                 reject_if: :all_blank
 
   validates :category_id, presence: true
   validates :content, presence: true, length: {maximum: 100}
+
+  private
+  def check_answer
+    if answers.select{|a| a.correct}.blank?
+      errors.add(:base, "You have to choose one correct answer.")
```
Comment (dieunb) [inline app/models/question.rb]: remove `()` and use i18n

File: app/views/admins/categories/show.html.erb [hunk 6]
```diff
@@ -6,4 +6,10 @@
   <div>
     <label>Description: </label><%= @category.description %>
   </div>
+  <div>
+  <%= link_to "Add Question", 
+  [:new, :admins, @category, :question],
+    # new_admins_category_question_path(@category),
```
Comment (dieunb) [inline app/views/admins/categories/show.html.erb]: remove remark

File: app/controllers/admins/questions_controller.rb (hunk 2 above)
Comment (kouch-chivorn) [inline app/controllers/admins/questions_controller.rb]: I use @category in the views: questions/new.html

File: app/helpers/application_helper.rb (hunk 4 above)
Comment (kouch-chivorn) [inline app/helpers/application_helper.rb]: OK, I will read and try to use it later.

**Pull request merged**